deepfield/impyla
# QP-3041 skip version checks, rebuild thrift files

Addresses [QP-3041](https://deepfield.atlassian.net/browse/QP-3041)

(The files to look at are _thrift_api.py, hiveserver2.py, and setup.py, the rest were modified by recompiling)

Impyla uses .py files compiled by thrift. At some point impyla switched to using thrift files directly instead of the compiled .py files due to incompatibility issues, but at a performance cost. Since then, thrift has been updated and can once again generate .py files that are usable, but we also need to bypass python version checks in a few places to force it to use these files the "old way".

We'll also need to add a conda recipe and make the appropriate systems changes to switch to using this fork, up to now we've been using the original cloudera version of impyla (note that this fork is at version 0.15.0, which matches the version of impyla we currently use in conda)

## How to Test
1. On a vlab, clone the deepfield impyla repo with `git clone git@github.com:deepfield/impyla.git`
2. Checkout this branch
3. First, check query performance before the change. In an ipython shell:
```
import deepy.ibis.util

with deepy.ibis.util.default_connection() as con:
    %time cur = con.raw_sql('select * from deepfield_traffic_step10.traffic limit 100000', results=True)
    %time cur.fetchall(columnar=True)
```
You should get something like:
```
CPU times: user 28 ms, sys: 4 ms, total: 32 ms
Wall time: 4.07 s
CPU times: user 39 s, sys: 456 ms, total: 39.5 s
Wall time: 40.1 s
```
4. "Install" these changes by copying the impala directory from impyla to the site-packes directory: `sudo cp -r impala/ /pipedream/local/venv/deepfield-env/lib/python3.7/site-packages/`
5. Repeat step 3, this time the user+sys time should go down by an order of magnitude, something like:
```
CPU times: user 20 ms, sys: 0 ns, total: 20 ms
Wall time: 3.78 s
CPU times: user 1.38 s, sys: 328 ms, total: 1.7 s
Wall time: 2.42 s
```
